### PR TITLE
Run akka-stream-tests on Scala 3

### PR DESCRIPTION
**Pull request by raboof**
Refs #30243

### CODE REVIEW

**Comment by raboof** [comment]
Failure was #30464

**Comment by raboof** [comment]
Failure was #30469

File: akka-stream-tests/src/test/scala/akka/stream/DslConsistencySpec.scala [hunk 1]
```diff
@@ -47,6 +47,7 @@ class DslConsistencySpec extends AnyWordSpec with Matchers {
     Set("equals", "hashCode", "notify", "notifyAll", "wait", "toString", "getClass") ++
     Set("productArity", "canEqual", "productPrefix", "copy", "productIterator", "productElement") ++
     Set("productElementName", "productElementNames") ++
+    Set("_1") ++
```
Comment (ignasi35) [inline akka-stream-tests/src/test/scala/akka/stream/DslConsistencySpec.scala:50]: Not sure where this `_1` comes from but I have an itch that tells me that if there's `_1`, then there'll be a `_2`...

File: akka-stream-tests/src/test/java/akka/stream/javadsl/GraphDslTest.java [hunk 2]
```diff
@@ -225,10 +224,13 @@ public void beAbleToConstructClosedGraphFromList() throws Exception {
...
-    Pair<NotUsed, NotUsed> pair = Source.fromGraph(mappedMatValueSrcGraph).to(snk).run(system);
-    assertEquals(pair, new Pair<>(NotUsed.getInstance(), NotUsed.getInstance()));
+    // FIXME it looks like the Graph.mapMaterializedValue static forwarder is no longer
+    // generated on Scala 3
+    //    Graph<SourceShape<Object>, Pair<NotUsed, NotUsed>> mappedMatValueSrcGraph =
+    //        Graph.mapMaterializedValue(srcGraph, notUsed -> new Pair<>(notUsed, notUsed));
+    //    Sink<Object, CompletionStage<Done>> snk = Sink.ignore();
+    //    Pair<NotUsed, NotUsed> pair =
+    // Source.fromGraph(mappedMatValueSrcGraph).to(snk).run(system);
+    //    assertEquals(pair, new Pair<>(NotUsed.getInstance(), NotUsed.getInstance()));
```
Comment (ignasi35) [inline akka-stream-tests/src/test/java/akka/stream/javadsl/GraphDslTest.java:234]: Not sure how to move on with this. Raise a separate issue and unblock this PR? Hold this PR until we find a solution?

Sounds like a blocker.

**Review by ignasi35 — APPROVED** [review]
LGTM, but there was a lot of repetition so it's likely I skimmed over the needle in the haystack of the PR.

File: akka-stream-tests/src/test/scala/akka/stream/DslConsistencySpec.scala (hunk 1 above)
Comment (raboof) [reply akka-stream-tests/src/test/scala/akka/stream/DslConsistencySpec.scala:50]: This test checks that methods that are available on javadsl are also available on the scaladsl and/or vice-versa. This collection lists methods that are allowed to exist on one but not the other. I haven't looked at the details, but apparently one of the classes now has a `_1` method on one but not the other. We haven't seen a test failure for a `_2`, so apparently not in this case ;).

File: akka-stream-tests/src/test/java/akka/stream/javadsl/GraphDslTest.java (hunk 2 above)
Comment (raboof) [reply akka-stream-tests/src/test/java/akka/stream/javadsl/GraphDslTest.java:234]: I'm not sure why it'd be a blocker: having a Scala 3 artifact without the `Graph.mapMaterializedValue` javadsl API method seems like an improvement over having no Scala 3 artifact at all ;), and it's still there on the 2.12 artifacts.

I'll create a proper issue though.

File: akka-stream-tests/src/test/scala/akka/stream/DslConsistencySpec.scala (hunk 1 above)
Comment (ignasi35) [reply akka-stream-tests/src/test/scala/akka/stream/DslConsistencySpec.scala:50]: We're a test away to find out. :D

Not a show-stopper. ;-)

File: akka-stream-tests/src/test/java/akka/stream/javadsl/GraphDslTest.java (hunk 2 above)
Comment (ignasi35) [reply akka-stream-tests/src/test/java/akka/stream/javadsl/GraphDslTest.java:235]: I saw you raised an issue and linked to it. Works for me.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**